src/diff.ml

```diff
@@ -0,0 +1,93 @@
+
+type diff =
+  { words: int
+  ; allocations: int
+  ; blocks: int
+  }
+
+let empty_diff = { allocations = 0; words = 0; blocks = 0 }
+
+module AllocMap = Map.Make(String)
+module NameSet = Set.Make(String)
+
+
+let get_alloc_map snapshot =
+  let rec loop snapshot acc =
+    Section.fold Section.(fun addr item acc ->
+      let value = Item.value item in
+
+      let words = Allocation.words value in
+      let allocations = Allocation.allocations value in
+      let blocks = Allocation.blocks value in
+
+      try
+        let display = Item.display item in
+        let name = String.sub display 0 (String.index display ':') in
+        acc |> AllocMap.update name (fun v -> Some (match v with
```

Comment:
I'd prefer to avoid using `|>` in general and especially for just passing an argument like this.